### ISSUE TITLE
Parsing does not parse as the default time zone

### ISSUE THREAD

**Issue by jeremyhaile** [moment/moment-timezone#197]
When i set a default time zone, moment is not returning the date I'm expecting.  

When parsing `2015-03-26` in America/Los_Angeles time zone I am expecting `Wed Mar 26 2015 00:00:00 GMT-0700` but instead all of the dates that come back are offset as if moment is parsing the date in browser time and then converting it to the default time zone after parsing.

```ruby
moment.tz('2015-03-26', 'America/Los_Angeles').toString() # Expected value: "Thu Mar 26 2015 00:00:00 GMT-0700"

# Set default and now values are not expected
moment.tz.setDefault('America/Los_Angeles') 
moment('2015-03-26').toString()           # "Wed Mar 25 2015 21:00:00 GMT-0700"
moment('2015-03-26 00:00').toString()     # "Wed Mar 25 2015 21:00:00 GMT-0700"
moment.tz('2015-03-26').toString()        # "Wed Mar 25 2015 21:36:31 GMT+0000"
moment.tz('2015-03-26 00:00').toString()  # "Wed Mar 25 2015 21:36:31 GMT+0000"

```

**Comment by jeremyhaile** [comment]
It's also strange that using moment-timezone to parse is also preserving my browser's minutes and seconds, even when I explicitly specify them.

**Comment by jeremyhaile** [comment]
Even if I explicitly specify the time zone, it returns an unexpected value. It is as if it is applying the default time zone twice.

```ruby
moment.tz('03-26-2015', 'America/Los_Angeles').toString() # "Wed Mar 25 2015 17:00:00 GMT-0700"
```

**Comment by maddoger** [comment]
I have the same problem. moment.js using local time for parsing and converting it to the right time zone.
For example: local time zone is +0400. 
```js
moment.tz.setDefault('Europe/Moscow'); //+0300
moment("09.04.2015 22:22", "DD.MM.YYYY HH:mm").toString(); //Thu Apr 09 2015 21:22:00 GMT+0300
```
Expected value is `Thu Apr 09 2015 22:22:00 GMT+0300`, but real value is `Thu Apr 09 2015 21:22:00 GMT+0300`.

**Comment by mj1856** [comment]
Verified.  I see the same behaviors.

**Comment by timrwood** [comment]
Some of these are bugs and some are not, so I'll narrow them down.

These are both bugs and are addressed in #217.

```js
moment('2015-03-26').toString()       // Wed Mar 25 2015 21:00:00 GMT-0700
moment('2015-03-26 00:00').toString() // Wed Mar 25 2015 21:00:00 GMT-0700
```

These use an incorrect signature for `moment.tz` and are not bugs. The correct signature has a timezone name as the last parameter. For example: `moment.tz('2015-03-26', 'America/Los_Angeles')`

```js
moment.tz('2015-03-26').toString()       // Wed Mar 25 2015 21:36:31 GMT+0000
moment.tz('2015-03-26 00:00').toString() // Wed Mar 25 2015 21:36:31 GMT+0000
```

This is the correct signature for `moment.tz` but the input follows the same rules as the `moment()` method, and inputs that do not have a format string and do not match ISO8601 fall through to the `Date` constructor and cannot be used to parse into a zone as noted [here](http://momentjs.com/timezone/docs/#/using-timezones/parsing-in-zone/). This is not a bug.

```js
moment.tz('03-26-2015', 'America/Los_Angeles').toString() // Wed Mar 25 2015 17:00:00 GMT-0700
```

This is also a bug and is addressed in #217.

```js
moment("09.04.2015 22:22", "DD.MM.YYYY HH:mm").toString(); // Thu Apr 09 2015 21:22:00 GMT+0300
```

**Comment by timrwood** [comment]
The fix in #217 is released in `0.4.0`.
